### PR TITLE
Added the test containers for the postgress

### PR DESCRIPTION
**Pull request by atharmajeed**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LASB-XXX)

Describe what you did and why.

## Checklist

Before you ask people to review this PR:

- [ ] Tests should be passing: `./gradlew test`
- [ ] Github should not be reporting conflicts; you should have recently run `git rebase main`.
- [ ] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [ ] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [ ] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: dces-drc-integration/src/test/java/uk/gov/justice/laa/crime/dces/integration/config/ApplicationTestConfig.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.testcontainers.service.connection.ServiceConnection;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+@Testcontainers
+@SpringBootTest
+public abstract class ApplicationTestConfig {
```
Comment (vvartakMoJ) [inline dces-drc-integration/src/test/java/uk/gov/justice/laa/crime/dces/integration/config/ApplicationTestConfig.java:11]: I think a better name for this class is ApplicationTestBase.

File: dces-drc-integration/src/test/java/uk/gov/justice/laa/crime/dces/integration/service/ContributionServiceTest.java [hunk 2]
```diff
@@ -58,11 +57,11 @@
 import static org.springframework.http.HttpStatus.OK;
 
 @ExtendWith(SoftAssertionsExtension.class)
-@SpringBootTest
-@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+//@SpringBootTest
+//@TestInstance(TestInstance.Lifecycle.PER_CLASS)
```
Comment (LAA-Dave-Edwards) [inline dces-drc-integration/src/test/java/uk/gov/justice/laa/crime/dces/integration/service/ContributionServiceTest.java:61]: Probably missed, but should remove these commented lines.

File: dces-drc-integration/src/test/java/uk/gov/justice/laa/crime/dces/integration/client/DrcApiClientTest.java [hunk 3]
```diff
@@ -49,15 +48,15 @@ class DrcApiClientTest {
     @Autowired
     private ServicesProperties services;
 
-    @BeforeAll
+    @BeforeEach
```
Comment (LAA-Dave-Edwards) [inline dces-drc-integration/src/test/java/uk/gov/justice/laa/crime/dces/integration/client/DrcApiClientTest.java:51]: Kicking off the webserver for each call is going to be costly, no? This just to clean out any residual data/modifications? If so is there a lower impact way?

Comment (vvartakMoJ) [reply dces-drc-integration/src/test/java/uk/gov/justice/laa/crime/dces/integration/client/DrcApiClientTest.java:51]: I'm guessing in Athar's absence, but it's supposed to be a light-weight operation and isolates any leftover requests on the mock server failing subsequent tests. We can revisit this if it becomes a problem in future.

File: dces-drc-integration/src/test/java/uk/gov/justice/laa/crime/dces/integration/config/MaatApiWebClientConfigurationTest.java [hunk 4]
```diff
@@ -23,29 +23,26 @@
...
+    private MockWebServer mockWebServer;
 
     @Autowired
     private ServicesProperties services;
 
     @MockBean
     OAuth2AuthorizedClientManager authorizedClientManager;
 
-    @BeforeAll
+    @BeforeEach
```
Comment (LAA-Dave-Edwards) [inline dces-drc-integration/src/test/java/uk/gov/justice/laa/crime/dces/integration/config/MaatApiWebClientConfigurationTest.java:45]: Same as other question.

File: dces-drc-integration/src/test/java/uk/gov/justice/laa/crime/dces/integration/controller/DcesDrcErrorControllerTest.java [hunk 5]
```diff
@@ -4,18 +4,18 @@
...
 import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
 import static uk.gov.justice.laa.crime.dces.integration.utils.RestTestUtils.getHttpHeaders;
 
-@SpringBootTest
+
 @AutoConfigureMockMvc
 @AutoConfigureObservability
-class DcesDrcErrorControllerTest {
+class DcesDrcErrorControllerTest extends ApplicationTestConfig {
```
Comment (LAA-Dave-Edwards) [inline dces-drc-integration/src/test/java/uk/gov/justice/laa/crime/dces/integration/controller/DcesDrcErrorControllerTest.java:18]: Is this going to need to be added to all our tests?

File: dces-drc-integration/src/test/java/uk/gov/justice/laa/crime/dces/integration/service/ContributionServiceTest.java (hunk 2 above)
Comment (vvartakMoJ) [reply dces-drc-integration/src/test/java/uk/gov/justice/laa/crime/dces/integration/service/ContributionServiceTest.java:61]: Removed the commented lines.

File: dces-drc-integration/src/test/java/uk/gov/justice/laa/crime/dces/integration/config/ApplicationTestConfig.java (hunk 1 above)
Comment (vvartakMoJ) [reply dces-drc-integration/src/test/java/uk/gov/justice/laa/crime/dces/integration/config/ApplicationTestConfig.java:11]: Making this change (executive decision in Athar's absence!).

**Review by LAA-Dave-Edwards — APPROVED** [review]
Looks good.

**Pull request merged**